File: scala-hoots/src/main/java/io/hoots/domain/Item.java
package io.hoots.domain;

import java.util.UUID;

/**
 * Created by rwadowski on 13.04.17.
 */
public class Item {

    private final String name;
    private final UUID id;

    public Item(String name) {
        this.name = name;
        this.id = UUID.randomUUID();
    }

    public String name() {
        return this.name;
    }

    public UUID id() {
        return id;
    }

    @Override
    public boolean equals(Object object) {
        if(null == object) {
            return false;
        }

        if(this == object) {
            return true;
        }

        if(!(object instanceof Item)) {
            return false;
        }

        final Item that = (Item) object;

        return this.name.equals(that.name)
            && this.id.equals(that.id);
    }

    @Override
    public String toString() {
        return "Item(" + this.name + ")";
    }
}

File: scala-hoots/src/main/scala/io/hoots/input/domain/Item.scala
package io.hoots.input.domain

import java.util.UUID

/**
  * Created by rwadowski on 19.04.17.
  */
case class Item(name: String, id: UUID) {

  def this(name: String) = this(name, UUID.randomUUID())
}

object Item {

  def apply(name: String): Item = new Item(name, UUID.randomUUID())
}